File: Node/Prisma_ORM/src/api/orderDetailAPI.js
const express = require('express');
const OrderDetailService = require('../bll/services/orderDetailService');
const router = express.Router();

router.get('/', async (req, res) => {
  try {
    const entities = await OrderDetailService.getAll();
    res.status(200).json(entities);
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});

router.get('/:orderID/:productID', async (req, res) => {
  const { orderID, productID } = req.query;
  try {
    const entity = await OrderDetailService.getById(Number(orderID), Number(productID));
    res.status(200).json(entity);
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});

router.post('/', async (req, res) => {
  const entity = req.body;
  try {
    res.status(200).json(await OrderDetailService.create(entity));
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});


router.delete('/:orderID/:productID', async (req, res) => {
  const { orderID, productID } = req.params;
  try {
    res.status(200).json(await OrderDetailService.delete(Number(orderID), Number(productID)));
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});

module.exports = router;

File: Go/GORM_ORM/api/orderDetailAPI.go
package api

import (
	"net/http"
	"strconv"

	"GORM_ORM/bll/services"
	"GORM_ORM/dal/entities"

	"github.com/gin-gonic/gin"
)

type OrderDetailAPI struct {
	Service services.OrderDetailService
}

func NewOrderDetailAPI(s services.OrderDetailService) *OrderDetailAPI {
	return &OrderDetailAPI{Service: s}
}

func (h *OrderDetailAPI) GetAll(c *gin.Context) {
	entities, err := h.Service.GetAll()
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}
	c.JSON(http.StatusOK, entities)
}

func (h *OrderDetailAPI) GetByID(c *gin.Context) {
	orderId, _ := strconv.Atoi(c.Param("orderId"))
	productId, _ := strconv.Atoi(c.Param("productId"))
	entity, err := h.Service.GetByID(orderId, productId)
	if err != nil {
		c.JSON(http.StatusNotFound, gin.H{"error": "OrderDetail not found"})
		return
	}
	c.JSON(http.StatusOK, entity)
}

func (h *OrderDetailAPI) Create(c *gin.Context) {
	var entity entities.OrderDetail
	if err := c.ShouldBindJSON(&entity); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}
	if err := h.Service.Create(&entity); err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}
	c.JSON(http.StatusCreated, entity)
}

func (h *OrderDetailAPI) Delete(c *gin.Context) {
	orderId, _ := strconv.Atoi(c.Param("orderId"))
	productId, _ := strconv.Atoi(c.Param("productId"))
	if err := h.Service.Delete(orderId, productId); err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}
	c.Status(http.StatusOK)
}
